test/transpile.jl

```diff
@@ -370,7 +370,72 @@ end
     for (input_gates,gates_in_output) in test_specs
         circuit = QuantumCircuit(qubit_count = qubit_count, gates=input_gates) 
     
+        transpiled_circuit=transpile(transpiler,circuit)
+        
+        gates=get_circuit_gates(transpiled_circuit)
+        
+        @test length(gates)==gates_in_output
         
+        if !(compare_circuits(circuit,transpiled_circuit))
+            println("gates in input: $(get_circuit_gates(circuit))")
+        end
+
+        @test compare_circuits(circuit,transpiled_circuit)
+
+    end
+
+end
+
+
+@testset "SwapQubitsForAdjacencyTranspiler{LatticeConnectivity}" begin
+    
+    nrows       =4
+    ncols       =3
+    qubit_count =nrows*ncols
+    transpiler = SwapQubitsForAdjacencyTranspiler(LatticeConnectivity(nrows, ncols))
+
+    # LatticeConnectivity{4,3}
+    #        1 ──  2 
+    #        |     | 
+    #  3 ──  4 ──  5 ──  6 
+    #        |     |     | 
+    #        7 ──  8 ──  9 ── 10 
+    #              |     | 
+    #             11 ── 12 
+
+    test_specs=[
+        # (gates list       gates_in_output)
+        ([swap(1,2)],       1)      # no effect
+        ([swap(2,8)],       3)
+        ([swap(8,2)],       3)
+
+        ([iswap(1,2)],       1)      # no effect
+        ([iswap(2,8)],       3)
+        ([iswap(8,2)],       3)
+
+        ([control_z(5,6)],  1)      # no effect
+        ([control_z(5,10)], 5)      # target at bottom
+        ([control_z(10,1)], 9)     # target on top
+
+        ([toffoli(4,5,6)],  1)      # no effect
```

Comment:
The adjacency transpilers do not have to deal with 3 qubit gates. We can remove these tests.

Reply:
I would rather leave them in even though we wont use it with the Sequential Transpiler, as the SwapForAdjacency stage can still accomodate them. If this feature is left in it should also have a test.